core/src/value/primitive.rs

```diff
@@ -3172,56 +2978,191 @@ impl PrimitiveValue {
+    ///             .ymd_opt(1992, 1, 1).unwrap()
+    ///             .and_hms_micro_opt(0, 0, 0, 0).unwrap()
+    ///         ),
+    ///         end: Some(Local::now().offset()
+    ///             .ymd_opt(1993, 12, 31).unwrap()
+    ///             .and_hms_micro_opt(23, 59, 59, 999_999).unwrap()
+    ///         )
+    ///     }
+    /// );
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn to_datetime_range(&self) -> Result<DateTimeRange, ConvertValueError> {
+        match self {
+            PrimitiveValue::DateTime(dt) if !dt.is_empty() => dt[0]
+                .range()
+                .context(ParseDateTimeRangeSnafu)
+                .map_err(|err| ConvertValueError {
+                    requested: "DateTimeRange",
+                    original: self.value_type(),
+                    cause: Some(err),
+                }),
+            PrimitiveValue::Str(s) => super::range::parse_datetime_range(s.trim_end().as_bytes())
+                .context(ParseDateTimeRangeSnafu)
+                .map_err(|err| ConvertValueError {
+                    requested: "DateTimeRange",
+                    original: self.value_type(),
+                    cause: Some(err),
+                }),
+            PrimitiveValue::Strs(s) => super::range::parse_datetime_range(
+                s.first().map(|s| s.trim_end().as_bytes()).unwrap_or(&[]),
+            )
+            .context(ParseDateTimeRangeSnafu)
+            .map_err(|err| ConvertValueError {
+                requested: "DateTimeRange",
+                original: self.value_type(),
+                cause: Some(err),
+            }),
+            PrimitiveValue::U8(bytes) => {
+                super::range::parse_datetime_range(trim_last_whitespace(bytes))
+                    .context(ParseDateTimeRangeSnafu)
+                    .map_err(|err| ConvertValueError {
+                        requested: "DateTimeRange",
+                        original: self.value_type(),
+                        cause: Some(err),
+                    })
+            }
+            _ => Err(ConvertValueError {
+                requested: "DateTimeRange",
+                original: self.value_type(),
+                cause: None,
+            }),
+        }
+    }
+
+    /// Retrieve a single `DateTimeRange` from this value.
+    ///
+    /// Use a custom ambiguous date-time range parser.
+    ///
+    /// For full description see [PrimitiveValue::to_datetime_range] and [AmbiguousDtRangeParser].
+    /// # Example
+    ///
+    /// ```
+    /// # use dicom_core::value::{C, PrimitiveValue};
+    /// # use std::error::Error;
+    /// use dicom_core::value::range::{AmbiguousDtRangeParser, ToLocalTimeZone, IgnoreTimeZone, FailOnAmbiguousRange, DateTimeRange};
+    /// use chrono::{NaiveDate, NaiveTime, NaiveDateTime};
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    ///
+    /// // The DICOM protocol allows for parsing text representations of date-time ranges,
```

Comment:
Maybe some of these explanations about DICOM date-time representations would fit better in the documentation corpus, rather than inside a doctest?